### PR TITLE
Add user_research_observation_agreed boolean to Registration

### PR DESCRIPTION
**Pull request by tvararu**
This will be used to track whether users expressing interest in the pilot consent to their child being observed as part of the session.

### Screenshot

Screenshot 2024-02-15 at 12 41 37

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -301,6 +302,7 @@
     t.boolean "terms_and_conditions_agreed"
     t.boolean "data_processing_agreed"
     t.boolean "consent_response_confirmed"
+    t.boolean "observed_session_agreed"
```
Comment (benilovj) [inline db/schema.rb:305]: I wonder if this might read better if it were `agreed_to_user_research_observation`

File: db/schema.rb [hunk 2]
```diff
@@ -235,6 +235,7 @@
     t.integer "team_id", null: false
     t.boolean "registration_open", default: false
     t.string "urn"
+    t.boolean "permission_to_observe_required"
```
Comment (benilovj) [inline db/schema.rb:238]: Could read slightly clearer if `involved_in_user_research_observation`
Very wordy though.

**Review by benilovj — COMMENTED** [review]
Very very optional naming suggestions, as the existing ones made me pause a minute to parse.

File: db/schema.rb (hunk 2 above)
Comment (misaka) [reply db/schema.rb:238]: Just my €0.02 ... I think `observed_session_agreed` -> `agreed_to_user_research_observation` gets us good value, more than this one.

**Review by misaka — APPROVED** [review]
:rocket: ... I think the first renaming would be good, the second would also help but a bit wordy.

File: db/schema.rb (hunk 1 above)
Comment (tvararu) [reply db/schema.rb:305]: `user_research_observation_agreed` matches the other fields which all end in `_agreed`, shall I do that instead?
